pkg/util/tail/tail_test.go

```diff
@@ -18,10 +18,45 @@ package tail
 
 import (
 	"bytes"
+	"io/ioutil"
+	"os"
 	"strings"
 	"testing"
 )
 
+func TestReadAtMost(t *testing.T) {
+	fakeFile, _ := ioutil.TempFile("", "")
+	defer os.Remove(fakeFile.Name())
+	fakeData := []byte("this is fake data")
+
+	ioutil.WriteFile(fakeFile.Name(), fakeData, 0600)
+
+	var readTests = []struct {
+		readLen int
+		out []byte
+		moreUnread bool
+	}{
+		{5, []byte(" data"), true},
+		{len(fakeData), fakeData, false},
+		{len(fakeData) + 1, fakeData, false},
```

Comment:
It doesn't seem so, the go docs on Seek do not cover this but I believe for fseek(3) this isn't considered an error.